Docker Image runs as root

# Bug Report

Rector runs as root. The permissions of all files are broken.

Overriding the user does not help:

```
❯ docker run --rm -v (pwd):/project -u1000 rector/rector:latest process src

 [ERROR] Unable to write in the "cache" directory (/tmp/_rector).     
```

I also ran into this problem. I want run the docker container as "me" and not root because rector changes the file permission of processed files to root (probably because the file is created from scratch and then copied back to the project) and after processing I can't write to the modified files anymore.

So, I just tried to run the command with `--user $(id -u):$(id -g)`:
`docker run --rm --user $(id -u):$(id -g) -v $(pwd):/project rector/rector:latest process src --config rector.php --dry-run`

But that throws an error `[ERROR] Unable to write in the "cache" directory (/tmp/_rector)`.

The first "solution" is to just ignoring the `/tmp` folder in the container with an additional parameter `-v /tmp:/tmp`:

`docker run --rm --user $(id -u):$(id -g) -v /tmp:/tmp -v $(pwd):/project rector/rector:latest process src --config rector.php --dry-run`.

That's working, but kind of ugly. So I tried to get to the root of the problem. Something in the container's `/tmp` file seems to cause issues:
```
$ docker run -it --rm --entrypoint "/bin/ls" rector/rector:latest -alh /tmp
total 36K
drwxrwxrwt 1 root root 4.0K Nov  4 14:56 .
drwxr-xr-x 1 root root 4.0K Nov 21 21:55 ..
drwxr-xr-x 3 root root 4.0K Nov  4 14:56 _rector
drwxr-xr-x 3 root root 4.0K Nov  4 14:56 _rector_cached_files
drwxr-xr-x 2 root root 4.0K Nov  4 14:56 _rector_log
drwxr-xr-x 2 root root 4.0K Nov  4 14:56 _rector_stubs
drwxr-xr-x 3 root root 4.0K Nov  4 14:56 cache
drwxr-xr-x 1 root root 4.0K Nov  4 14:56 opcache
drwxr-xr-x 5 root root 4.0K Oct 30 11:59 pear
```

The folder is not empty and the directory `/tmp/_rector` already exists and is owned by root.
That's because in the [Dockerfile rector is already being executed](https://github.com/rectorphp/rector/blob/master/Dockerfile#L38).
I haven't tried it yet, but probably removing the contents of the `/tmp` folder or `chmod`ing everything to `777` in the `Dockerfile` would solve that issue.

Thanks!

```
/tmp/bla via 🐘 v7.4.11 took 18s
❯ docker run --rm -u1000:1000 -v (pwd):/project rector/rector:fix-4568 process index.php

 [ERROR] Unable to create file
         '/tmp/_rector_stubs/7ba983888879c5dfc11be38815c040e2.php.lock'.
         fopen(/tmp/_rector_stubs/7ba983888879c5dfc11be38815c040e2.php.lock):
         failed to open stream: Permission denied
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -37,6 +37,8 @@ COPY . .
 # To warmup opcache a little
 RUN bin/rector list
 
+RUN chmod 777 -R /tmp
+
 ENTRYPOINT [ "rector" ]
 
 ENV PATH /rector/bin:$PATH

--- a/docs/how_to_run_rector_in_docker.md
+++ b/docs/how_to_run_rector_in_docker.md
@@ -16,3 +16,10 @@ docker run --rm -v $(pwd):/project rector/rector:latest process src \
     --config rector.php \
     --dry-run
 ```
+
+## Permissions issues
+
+If you run into issues with `permission denied` or running Rector in docker keeps changing owner of your project files, running container as current user `--user $(id -u):$(id -g)` should solve it for you:
+```
+docker run --rm --user $(id -u):$(id -g) -v $(pwd):/project rector/rector process src --config rector.php --dry-run
+```

